query-node/mappings/src/bounty.ts

```diff
@@ -0,0 +1,496 @@
+  store: DatabaseManager,
+  bountyId: BountyId,
+  contributor: string | undefined
+): Promise<BountyContribution> {
+  const contribution = await store.get(BountyContribution, { where: { bountyId, contributor } })
+  if (!contribution) {
+    const actorType = typeof contributor === 'undefined' ? 'council' : `member id ${contributor}`
+    throw new Error(`Bounty contribution not found by contributor: ${actorType}`)
+  }
+  return contribution
+}
+
+async function getEntry(store: DatabaseManager, entryId: EntryId): Promise<BountyEntry> {
+  const entry = await store.get(BountyEntry, { where: { id: entryId } })
+  if (!entry) {
+    throw new Error(`Entry not found by id: ${entryId}`)
+  }
+  return entry
+}
+
+async function updateBounty(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  bountyId: BountyId,
+  changes: (bounty: Bounty) => Partial<Bounty>
+) {
+  const bounty = await getBounty(store, bountyId)
+  bounty.updatedAt = new Date(event.blockTimestamp)
+  Object.assign(bounty, changes(bounty))
+
+  await store.save<Bounty>(bounty)
+
+  return bounty
+}
+
+async function updateEntry(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  entryId: EntryId,
+  changes: (entry: BountyEntry) => Partial<BountyEntry>
+) {
+  const entry = await getEntry(store, entryId)
+  entry.updatedAt = new Date(event.blockTimestamp)
+  Object.assign(entry, changes(entry))
+
+  await store.save<BountyEntry>(entry)
+
+  return entry
+}
+
+function bountyActorToMembership(actor: BountyActor): Membership | undefined {
+  if (actor.isMember) {
+    return new Membership({ id: String(actor.asMember) })
+  }
+}
+
+export function scheduledFundingEnd(bounty: Bounty, creationBlock: number): number | undefined {
+  if ('fundingPeriod' in bounty.fundingType) {
+    return creationBlock + bounty.fundingType.fundingPeriod
+  }
+}
+
+function scheduleBountyStageEnd(
+  stage: BountyStage,
+  endStage: (store: DatabaseManager, bounty: Bounty, blockNumber: number) => Promise<void>,
+  relations: string[] = []
+): (bounty: Bounty, blockNumber: number | undefined) => void {
+  return (bounty, blockNumber) => {
+    if (bounty.stage === stage && typeof blockNumber !== 'undefined') {
+      const bountyId = bounty.id
+
+      scheduleAtBlock(blockNumber, async (store) => {
+        const bounty = await getBounty(store, bountyId, relations)
+
+        if (bounty.stage === stage) {
+          await endStage(store, bounty, blockNumber)
+        }
+      })
+    }
+  }
+}
+
+function whenDef<T, R>(value: T | null | undefined, fn: (value: T) => R): R | undefined {
+  if (value !== null && typeof value !== 'undefined') return fn(value)
+}
+
+/**
+ * Schedule Periods changes
+ */
+
+export const bountyScheduleFundingEnd = scheduleBountyStageEnd(
+  BountyStage.Funding,
+  async (store, bounty, fundingPeriodEnd) => {
+    if ('minFundingAmount' in bounty.fundingType) {
+      const isFunded = bounty.totalFunding.gte(bounty.fundingType.minFundingAmount)
```

Comment:
Unfortunately looks like there is some typings bug in hydra and `bounty.fundingType.minFundingAmount` is not actually a `BN` (once fetched from the database), but a `string` instead. This is another issue on the long list of issues to be fixed in Hydra (hopefully most of them will be gone once we migrate to Subsquid)

When comparing a BN to a string you will get unexpected results.

A temporary workaround:
```suggestion
      const isFunded = bounty.totalFunding.gte(new BN(bounty.fundingType.minFundingAmount))
```